<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
  "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>4.1. Motivating example: visualizing viral evolution#</title>
<link href="../Styles/Style.css" type="text/css" rel="stylesheet"/>
</head>
<body>
<h1>4.1. Motivating example: visualizing viral evolution#</h1>
<blockquote>原文：<a href="https://mmids-textbook.github.io/chap04_svd/01_motiv/roch-mmids-svd-motiv.html">https://mmids-textbook.github.io/chap04_svd/01_motiv/roch-mmids-svd-motiv.html</a></blockquote>

<p><strong>Figure:</strong> Helpful map of ML by scitkit-learn (<a class="reference external" href="https://scikit-learn.org/stable/tutorial/machine_learning_map/index.html">Source</a>)</p>
<p><img alt="ml-cheat-sheet" src="../Images/19ac9e49b2f297976e40fee63e1c4ba0.png" data-original-src="https://scikit-learn.org/1.4/_static/ml_map.png"/></p>
<p><span class="math notranslate nohighlight">\(\bowtie\)</span></p>
<p>We consider an application of dimensionality reduction in biology. We will look at single-nucleotide polymorphism (SNP)<span class="math notranslate nohighlight">\(\idx{single-nucleotide polymorphism}\xdi\)</span> data from viruses. A little background first. From <a class="reference external" href="https://en.wikipedia.org/wiki/Single-nucleotide_polymorphism">Wikipedia</a>:</p>
<blockquote>
<div><p>A single-nucleotide polymorphism (SNP) is a substitution of a single nucleotide that occurs at a specific position in the genome, where each variation is present at a level of more than 1% in the population. For example, at a specific base position in the human genome, the C nucleotide may appear in most individuals, but in a minority of individuals, the position is occupied by an A. This means that there is a SNP at this specific position, and the two possible nucleotide variations – C or A – are said to be the alleles for this specific position.</p>
</div></blockquote>
<p>Quoting <a class="reference external" href="https://bmcgenet.biomedcentral.com/articles/10.1186/1471-2156-11-94">Jombart et al., BMC Genetics (2010)</a>, we analyze:</p>
<blockquote>
<div><p>the population structure of seasonal influenza A/H3N2 viruses using hemagglutinin (HA) sequences. Changes in the HA gene are largely responsible for immune escape of the virus (antigenic shift), and allow seasonal influenza to persist by mounting yearly epidemics peaking in winter. These genetic changes also force influenza vaccines to be updated on a yearly basis. […] Assessing the genetic evolution of a pathogen through successive epidemics is of considerable epidemiological interest. In the case of seasonal influenza, we would like to ascertain how genetic changes accumulate among strains from one winter epidemic to the next.</p>
</div></blockquote>
<p>Some details about the Jombart et al. dataset:</p>
<blockquote>
<div><p>For this purpose, we retrieved all sequences of H3N2 hemagglutinin (HA) collected between 2001 and 2007 available from Genbank. Only sequences for which a location (country) and a date (year and month) were available were retained, which allowed us to classify strains into yearly winter epidemics.</p>
</div></blockquote>
<p>We load a dataset, which contains a subset of strains from the dataset mentioned above.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span/><span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">'h3n2-snp.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>This is a large dataset. Here are the first five rows and first 10 colums.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span/><span class="nb">print</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="mi">5</span><span class="p">,</span> <span class="p">:</span><span class="mi">10</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span/>     strain  s6a  s6c  s6g  s17a  s17g  s17t  s39a  s39c  s39g
0  AB434107  1.0  0.0  0.0   1.0   0.0   0.0   0.0   0.0   1.0
1  AB434108  1.0  0.0  0.0   1.0   0.0   0.0   0.0   0.0   1.0
2  CY000113  1.0  0.0  0.0   1.0   0.0   0.0   0.0   0.0   1.0
3  CY000209  1.0  0.0  0.0   1.0   0.0   0.0   0.0   0.0   1.0
4  CY000217  1.0  0.0  0.0   1.0   0.0   0.0   0.0   0.0   1.0
</pre></div>
</div>
</div>
</div>
<p>For positions <code class="docutils literal notranslate"><span class="pre">6</span></code>, <code class="docutils literal notranslate"><span class="pre">17</span></code>, <code class="docutils literal notranslate"><span class="pre">39</span></code>, etc., the corresponding columns indicate which nucleotide (<code class="docutils literal notranslate"><span class="pre">a</span></code>, <code class="docutils literal notranslate"><span class="pre">c</span></code>, <code class="docutils literal notranslate"><span class="pre">g</span></code>, <code class="docutils literal notranslate"><span class="pre">t</span></code>) is present in the strain with a <code class="docutils literal notranslate"><span class="pre">1.0</span></code>. For example, strain <code class="docutils literal notranslate"><span class="pre">AB434107</span></code> has an <code class="docutils literal notranslate"><span class="pre">a</span></code> at position <code class="docutils literal notranslate"><span class="pre">6</span></code> and <code class="docutils literal notranslate"><span class="pre">17</span></code>, and a <code class="docutils literal notranslate"><span class="pre">g</span></code> at position <code class="docutils literal notranslate"><span class="pre">39</span></code>.</p>
<p>Overall it contains <span class="math notranslate nohighlight">\(1642\)</span> strains (whose names are listed in the first colum). The data lives in a <span class="math notranslate nohighlight">\(317\)</span>-dimensional space (not counting the name of strain, i.e., the first column).</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span/><span class="n">data</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span/>(1642, 318)
</pre></div>
</div>
</div>
</div>
<p>Obviously, vizualizing this data is not straighforward. How can we make sense of it? More specifically, how can we explore any underlying structure it might have. Quoting <a class="reference external" href="https://en.wikipedia.org/wiki/Exploratory_data_analysis">Wikipedia</a>:</p>
<blockquote>
<div><p>In statistics, exploratory data analysis (EDA) is an approach of analyzing data sets to summarize their main characteristics, often using statistical graphics and other data visualization methods. […] Exploratory data analysis has been promoted by John Tukey since 1970 to encourage statisticians to explore the data, and possibly formulate hypotheses that could lead to new data collection and experiments.</p>
</div></blockquote>
<p>In this chapter we will encounter an important mathematical technique for dimension reduction, which allow us to explore this data – and find interesting structure – in <span class="math notranslate nohighlight">\(2\)</span> (rather than <span class="math notranslate nohighlight">\(317\)</span>!) dimensions.</p>
    
</body>
</html>